lib/components/recordingSession/recorder.dart

```diff
@@ -28,94 +31,134 @@ class Recorder extends StatefulWidget {
+  String savePath;
+
+  int sessionCounter = 0;
+
+  Future<void> init() async {
+    tempDir = await getApplicationDocumentsDirectory();
+    recorderModule = FlutterSoundRecorder();
+  }
 
   @override
   void initState() {
     super.initState();
-    flutterSound = FlutterSound();
-    flutterSound.setSubscriptionDuration(0.01);
-    flutterSound.setDbPeakLevelUpdate(0.8);
-    flutterSound.setDbLevelEnabled(true);
+    init();
     initializeDateFormatting();
     userId = widget.userId;
     ws = widget.ws;
   }
 
+  Future<void> startAudioSession() async {
+    recorderModule.openAudioSession(
+        focus: AudioFocus.requestFocusTransient,
+        category: SessionCategory.playAndRecord,
+        mode: SessionMode.modeDefault,
+        device: AudioDevice.speaker);
+  }
+
+  Future<void> cancelRecorderSubscriptions() async {
+    if (_recorderSubscription != null) {
+      _recorderSubscription.cancel();
+      _recorderSubscription = null;
+    }
+  }
+
+  Future<void> closeAudioSession() async {
+    try {
+      await recorderModule.closeAudioSession();
+    } catch (e) {
+      print('Released unsuccessful');
+      print(e);
+    }
+  }
+
   @override
   void dispose() {
+    cancelRecorderSubscriptions();
+    closeAudioSession();
     super.dispose();
   }
 
   Future<void> startRecorder() async {
     ws.sendFirstMessage(userId);
+
+    final String now =
+        DateFormat('yyyy-MMMM-dd_HH:mm:ss:SSS').format(DateTime.now());
     try {
-      final DateTime now = DateTime.now();
-      final String date = DateFormat('yyyy-MM-ddThh:mm').format(now);
-      final Directory tempDir = await getTemporaryDirectory();
-      setState(() {
-        directory = tempDir;
-      });
-      print('tempdir: $tempDir');
-      final String path = await flutterSound.startRecorder(
-          uri: tempDir.path +
-              '/' +
-              date.toString() +
-              '_' +
-              userId.toString() +
-              '_sound.aac',
-          codec: t_CODEC.CODEC_AAC);
-      print('path: $path');
-      _recorderSubscription = flutterSound.onRecorderStateChanged.listen(
-        (RecordStatus e) {
-          final DateTime date = DateTime.fromMillisecondsSinceEpoch(
-              e.currentPosition.toInt(),
-              isUtc: true);
-          final String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
-          setState(
-            () {
-              _recorderTxt = txt.substring(0, 8);
-            },
-          );
-        },
+      await startAudioSession();
+      final PermissionStatus status = await Permission.microphone.request();
+      if (status != PermissionStatus.granted) {
+        throw RecordingPermissionException('Microphone permission not granted');
+      }
+
+      final String path =
+          '${tempDir.path}/${now}_user_${userId}_recording${ext[_codec.index]}';
+
+      await recorderModule.startRecorder(
+        toFile: path,
+        codec: _codec,
+        bitRate: 16000,
```

Comment:
i'm pretty sure we need the bit rate to be 16 (https://medium.com/@MicroPyramid/understanding-audio-quality-bit-rate-sample-rate-14286953d71f)